docker/Dockerfile

```diff
@@ -0,0 +1,81 @@
+FROM ubuntu:latest
+
+ADD sources.list /etc/apt/
+
+RUN apt-get update
+
+RUN DEBIAN_FRONTEND=noninteractive apt install -y tzdata
+
+# General dependencies
+RUN apt-get install -y \
+  git \
+  mercurial 
+RUN apt-get install -y \
+  wget \
+  vim \
+  autoconf \
+  bzr \
+  cvs 
+RUN apt-get install -y \
+  unrar \
+  build-essential \
+  clang \
+  valgrind \
+  gsl-bin 
+RUN apt-get install -y \
+  libgslcblas0 \
+  libgsl-dev \
+  flex \
+  bison \
+  libfl-dev \
+  tcpdump \
+  sqlite 
+RUN apt-get install -y \
+  sqlite3 \
+  libsqlite3-dev \
+  libxml2 \
+  libxml2-dev \
+  vtun \
+  lxc
+
+# QT4 components
+RUN apt-get install -y \
+  qtbase5-dev
+
+# Python components
+RUN apt-get install -y \
+  python \
+  python-dev \
+  python-setuptools \
+  cmake \
+  libc6-dev \
+  libc6-dev-i386 \
+  g++-multilib
+
+# NS-3
+
+# Create working directory
+RUN mkdir -p /usr/ns3
+WORKDIR /usr
+
+# Fetch NS-3 source
+RUN wget http://www.nsnam.org/release/ns-allinone-3.30.tar.bz2
+RUN tar -xf ns-allinone-3.30.tar.bz2
+
+# Configure and compile NS-3
+RUN cd ns-allinone-3.30&& ./build.py --enable-examples --enable-tests
```

Comment:
Instead of using ns-allinone-3.30 version, can we try to use git to support different versions?

Reply:
Yes, you can use "RUN git clone ns3" instead of "wget ns3.tar"

but I don't find the repo of ns3, I'm an ns3 beginner.

Reply:
NS3 repository is hosted on gitlab: [https://gitlab.com/nsnam/ns-3-dev](https://gitlab.com/nsnam/ns-3-dev).